pkg/server/serverpb/admin.proto

```diff
@@ -514,6 +514,40 @@ message QueryPlanResponse {
   string distsql_physical_query_plan = 1 [(gogoproto.customname) = "DistSQLPhysicalQueryPlan"];
 }
 
+message ReplicaMatrixRequest {
+}
+
+message ReplicaMatrixResponse {
+  message ZoneConfig {
+    // cli_specifier is ".default" for the default zone config.
+    // Otherwise, it's of the form [database.]table[@index[.partition]]
+    string cli_specifier = 1;
+
+    config.ZoneConfig config = 2 [(gogoproto.nullable) = false];
+
+    // config_yaml is the YAML representation of config.
+    string config_yaml = 3;
+  }
+
+  message TableInfo {
+    map<int32, int64> replica_count_by_node_id = 2
+      [(gogoproto.castkey) = "github.com/cockroachdb/cockroach/pkg/roachpb.NodeID"];
+
+    int64 zone_config_id = 3;
```

Comment:
@benesch this is the PR I mentioned IRL today. Partitions would slot in here, under tables. PTAL at this PR when you can; would like to get it and the accompanying frontend PR in soon, now that we're out of code freeze!